src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -103,16 +103,13 @@ public boolean canSwitchToBroker() {
         // authenticator
         // 4- signature of the broker is valid
         // 5- account exists
-        return AuthenticationSettings.INSTANCE.getUseBroker()
-                && verifyManifestPermissions()
+        return AuthenticationSettings.INSTANCE.getUseBroker() && verifyManifestPermissions()
                 && checkAccount(mAcctManager, "", "")
```

Comment:
1) if it's old broker, then there has to be one account existed 2) or it's the new broker which has the support to add user through broker.